### PR TITLE
Add JINJA_TESTS to use custom Jinja2 tests. Refs #1498

### PR DESCRIPTION
**Pull request by oseledets**
Sorry if I did anything wrong (first commit), but I tried to do a really simple thing that I use for my website, JINJA_TESTS settings added.

### CODE REVIEW

**Comment by ingwinlu** [comment]
lacking tests

**Comment by oseledets** [comment]
Found no tests for JINJA_FILTERS which has similar functionality. Will have to read how it is done.

**Comment by ingwinlu** [comment]
then you can add tests for JINJA_FILTERS as well <3

**Comment by ingwinlu** [comment]
https://github.com/getpelican/pelican/pull/1196

File: docs/settings.rst [hunk 1]
```diff
@@ -94,6 +94,7 @@ Setting name (followed by default value, if any)
                                                                                  control data. For example: ``[".hg", ".git", ".bzr"]``
 ``JINJA_EXTENSIONS = []``                                                        A list of any Jinja2 extensions you want to use.
 ``JINJA_FILTERS = {}``                                                           A dictionary of custom Jinja2 filters you want to use.
+``JINJA_TESTS = {}``                                                             A dictionary of custom Jinja2 tests you want to use
```
Comment (ametaireau) [inline docs/settings.rst]: Adding a link to http://jinja.pocoo.org/docs/dev/api/#writing-tests can help people getting started with Jinja2 tests.

**Comment by justinmayer** [comment]
Hi Ivan. Would you be willing to make the suggested changes, run tests locally to ensure they pass, and then push a squashed commit to this branch? Looking forward to getting this merged. (^_^)

**Comment by justinmayer** [comment]
Hi Ivan. Any chance you could take another pass at your pull request so we can get this merged?

**Comment by justinmayer** [comment]
@oseledets: Would you please let us know if you intend to finish your work on your contribution in this PR?

**Comment by nurupo** [comment]
I see this PR is supposed to replace the abandoned https://github.com/getpelican/pelican/pull/1196. Would be nice if you could also add support for the `JINJA_GLOBALS` from the original PR.

**Comment by nurupo** [comment]
Oh, never mind, I guess this PR is also abandoned.

**Comment by artsobolev** [comment]
Jeez, guys, can't believe you argue over adding 3 damn lines for almost 5 years by now!

**Comment by avaris** [comment]
I don't see any arguing. Some unit tests are requested and original author never came back to add them. You know, anybody can pick this up and finish it if one really wants this feature. Wonders of open source...

**Comment by artsobolev** [comment]
@nurupo, kudos to you! I initially hacked a custom filter, but your solution is way more elegant, switched to that one.

@avaris, proposed solution basically copies the `JINJA_FILTERS` implementation, that lacked any tests (and still does). And then the author was asked to implement them as well. Wonders, indeed.

**Comment by avaris** [comment]
If you want to finish this up and need any help along the way, let me know. I'll gladly assist.

**Comment by justinmayer** [comment]
I implemented this via the above-linked PR, which has been merged to master. 💫

**Comment by CNG** [comment]
Hey I picked a great time to pick up Pelican and find myself needing a custom Jinja test (to probably inefficiently create a list of unique tags across a subset of pages having a particular folder name in the path... still learning my options :-) ).

**Pull request closed without merging**